### PR TITLE
[Branch avro_1_9] Avro 1.9 upgrade compatible change - replaced deprecated public APIs with the compatible APIs

### PR DESCRIPTION
**Pull request by abhisheknath2011**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-XXX

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
- [ ] Replaced deprecated public APIs with the ones compatible with Avro 1.9.
- [ ] Avro 1.9 upgrade compatible change - Replaced guava library import from avro shaded with direct guava libraries.

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java [hunk 1]
```diff
@@ -147,9 +146,10 @@ public static Schema setSchemaCreationTime(Schema inputSchema, String creationTi
    * Common use cases for this method is in traversing {@link Schema} object into nested level and create {@link Schema}
    * object for non-root level.
    */
-  public static void convertFieldToSchemaWithProps(Map<String,JsonNode> fieldProps, Schema targetSchemaObj) {
-    for (Map.Entry<String, JsonNode> stringJsonNodeEntry : fieldProps.entrySet()) {
-      targetSchemaObj.addProp(stringJsonNodeEntry.getKey(), stringJsonNodeEntry.getValue());
+  public static void convertFieldToSchemaWithProps(Map<String,Object> fieldProps,
```
Comment (autumnust) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java:149]: Shall we preserve the original method still since it is a public static method ?

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java [hunk 2]
```diff
@@ -253,8 +259,16 @@ public GobblinYarnAppLauncher(Config config, YarnConfiguration yarnConfiguration
     YarnHelixUtils.setYarnClassPath(config, this.yarnConfiguration);
     YarnHelixUtils.setAdditionalYarnClassPath(config, this.yarnConfiguration);
     this.yarnConfiguration.set("fs.automatic.close", "false");
-    this.yarnClient = YarnClient.createYarnClient();
-    this.yarnClient.init(this.yarnConfiguration);
+    this.originalYarnRMAddress = this.yarnConfiguration.get(GobblinYarnConfigurationKeys.YARN_RESOURCE_MANAGER_ADDRESS);
```
Comment (autumnust) [inline gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java:262]: this doesn't seem to be relevant to your PR. Is this accidentally being brought in ? Can you rebase from the upstream if that's the case?

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnConfigurationKeys.java [hunk 3]
```diff
@@ -37,6 +37,8 @@
   public static final int DEFAULT_RELEASED_CONTAINERS_CACHE_EXPIRY_SECS = 300;
   public static final String APP_VIEW_ACL = GOBBLIN_YARN_PREFIX + "appViewAcl";
   public static final String DEFAULT_APP_VIEW_ACL = "*";
+  public static final String YARN_RESOURCE_MANAGER_ADDRESS = "yarn.resourcemanager.address";
```
Comment (autumnust) [inline gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnConfigurationKeys.java:40]: same as above comments about relevance.

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroSchemaUtils.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  private AvroSchemaUtils() {
+
+  }
+
+  public static Integer getValueAsInteger(final Schema schema, String prop) {
+    Object value = schema.getObjectProp(prop);
+    if(value instanceof Integer) {
+      return (Integer) value;
+    } else if(value instanceof String) {
```
Comment (autumnust) [inline gobblin-utility/src/main/java/org/apache/gobblin/util/AvroSchemaUtils.java:35]: space between `(` ? 
BTW you could auto-format the whole thing once you have style sheet set up in your IDE: https://gobblin.readthedocs.io/en/latest/developer-guide/CodingStyle/

File: gobblin-core-base/src/main/java/org/apache/gobblin/test/SequentialTestSource.java [hunk 5]
```diff
@@ -29,7 +29,7 @@
 import com.typesafe.config.Config;
 import com.typesafe.config.ConfigFactory;
 
-import avro.shaded.com.google.common.base.Throwables;
+import com.google.common.base.Throwables;
```
Comment (autumnust) [inline gobblin-core-base/src/main/java/org/apache/gobblin/test/SequentialTestSource.java:32]: are you sure this change is backward compatible ?

File: gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java [hunk 6]
```diff
@@ -632,7 +632,7 @@ private void addOrAlterPartitionWithPullMode(IMetaStoreClient client, Table tabl
             onPartitionExist(client, table, partition, nativePartition, existedPartition);
           }
         }
-      } catch (TException e) {
+      } catch (NoSuchObjectException e) {
```
Comment (autumnust) [inline gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java:635]: is this relevant ?

**Review by autumnust — CHANGES_REQUESTED** [review]

File: gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java (hunk 2 above)
Comment (abhisheknath2011) [reply gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java:262]: Please ignore this as this came from rebase. This change was introduced by one of the first three commits in the master branch. These three commits are not available in avro_1_9 and got added after rebase from upstream master.

File: gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java (hunk 6 above)
Comment (abhisheknath2011) [reply gobblin-hive-registration/src/main/java/org/apache/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java:635]: I think this is introduced by other three commit. Can you please change the changes specific to last two avro specific commits.

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java (hunk 1 above)
Comment (abhisheknath2011) [reply gobblin-utility/src/main/java/org/apache/gobblin/util/AvroUtils.java:149]: If we keep two methods one with jsonNode and other with Object, at some point we have to clean this existing method. For example, once we upgrade to Jackson 2.x by default 1.x classes need to be removed. Need to keep the Jackson 1.x classes wherever it is absolutely necessary. For example hive-serde and helix jar has Jackson 1.x dependency. So it would be good to remove this method and it would be helpful the fix the avro changes in a cleaner way and should be able detect issues (if there any) during testing.

File: gobblin-utility/src/main/java/org/apache/gobblin/util/AvroSchemaUtils.java (hunk 4 above)
Comment (abhisheknath2011) [reply gobblin-utility/src/main/java/org/apache/gobblin/util/AvroSchemaUtils.java]: Thanks for pointing this out. Applied Gobblin codestyle formatting.

File: gobblin-core-base/src/main/java/org/apache/gobblin/test/SequentialTestSource.java (hunk 5 above)
Comment (autumnust) [reply gobblin-core-base/src/main/java/org/apache/gobblin/test/SequentialTestSource.java:32]: Sounds good.

**Review by autumnust — APPROVED** [review]
Reviewed the last 2 commits.
